tools/rccl-prim-test/rccl_prim_test.cpp

```diff
@@ -122,21 +142,83 @@ do {                                                                           \
   }                                                                            \
 } while (0)
 
-static void setupPeers() {
-    int deviceCnt, dev;
-
-     HIPCHECK(hipGetDeviceCount(&deviceCnt));
-     HIPCHECK(hipGetDevice(&dev));
-    //! If gpus are not peer enabled, enable them
-    for (int i = 0; i < deviceCnt; i++) {
-         HIPCHECK(hipSetDevice(i));
-        for (int j = 0; j < deviceCnt; j++) {
-            if (i != j) {
-                HIPCHECK(hipDeviceEnablePeerAccess(j, 0));
-            }
+static void setupPeers(uint32_t *info) {
+  int deviceCnt, dev;
+
+  HIPCHECK(hipGetDeviceCount(&deviceCnt));
+  HIPCHECK(hipGetDevice(&dev));
+  //! If gpus are not peer enabled, enable them
+  for (int i = 0; i < deviceCnt; i++) {
+    HIPCHECK(hipSetDevice(i));
+    for (int j = 0; j < deviceCnt; j++) {
+      if (i != j) {
+	int p2p;
+        HIPCHECK(hipDeviceCanAccessPeer(&p2p, i, j));
+        if (!p2p) {
+          printf("Cannot enable peer access between device %d and %d. You may use HIP_VISIBLE_DEVICES to limit GPUs.\n",
+           i, j);
+          exit(-1);
         }
+        HIPCHECK(hipDeviceEnablePeerAccess(j, 0));
+        uint32_t linktype;
+        HIPCHECK(hipExtGetLinkTypeAndHopCount(i, j, &linktype, &info[i*deviceCnt+j]));
+      }
+      else
+        info[i*deviceCnt+j] = 0;
+    }
+  }
+  HIPCHECK(hipSetDevice(dev));
+}
+
+static void printRing(int id, int *ring, int deviceCnt) {
+  printf("Ring %d: ", id);
+  for (int i = 0; i < deviceCnt; i++)
+    printf("%1d ", ring[i]);
+  printf("\n");
+}
+
+static void findConnect(uint32_t *info, int *ring, int deviceCnt) {
+  int n = 0, curr = 0, best;
+  uint32_t temp[MAX_GPU*MAX_GPU];
+  for (int i = 0; i < deviceCnt*deviceCnt; i++) temp[i] = 0;
+  for (int i = 0; i < deviceCnt; i++) {
```

Comment:
Why do we loop over i here?  It's not used.

Reply:
This line?
for (int i = 0; i < deviceCnt*deviceCnt; i++) temp[i] = 0;
It is used to initialize the temp matrix to all 0s.

Reply:
Nevermind - That's the ring size